torch/csrc/stable/stableivalue_conversions.h

```diff
@@ -192,6 +193,37 @@ struct FromImpl<torch::stable::Tensor> {
   }
 };
 
+// Specialization for torch::headeronly::HeaderOnlyArrayRef<T> => StableIValue
+// Returns a new owning reference of the underlying list.
+template <typename T>
+struct FromImpl<torch::headeronly::HeaderOnlyArrayRef<T>> {
+  static StableIValue call(
+      const torch::headeronly::HeaderOnlyArrayRef<T>& val,
+      [[maybe_unused]] uint64_t extension_build_version,
+      [[maybe_unused]] bool is_internal) {
+    StableListHandle new_list_handle;
+    TORCH_ERROR_CODE_CHECK(
+        torch_new_list_reserve_size(val.size(), &new_list_handle));
+    for (const auto& elem : val) {
+      TORCH_ERROR_CODE_CHECK(torch_list_push_back(new_list_handle, from(elem)));
+    }
```

Comment:
you're going to leak memory for the list and for previously-pushed elements if this TORCH_ERROR_CODE_CHECK ever fails, courtesy of StableIValue not having a destructor